### PR TITLE
Browser: Add some conditional for url input from tab

### PR DESCRIPTION
**Pull request by ardi0x01**
adding some conditional if there is not contain top domain level then add it

### CODE REVIEW

**Comment by sin-ack** [comment]
This PR seems to mash multiple unrelated changes into a single commit. Please separate them and then force push (do not close this PR and open a new one).

**Review by sin-ack — CHANGES_REQUESTED** [review]

File: Userland/Applications/Browser/Tab.cpp [hunk 1]
```diff
@@ -408,12 +408,17 @@ Optional<URL> Tab::url_from_location_bar(MayAppendTLD may_append_tld)
 
     StringBuilder builder;
     builder.append(text);
+
     if (may_append_tld == MayAppendTLD::Yes) {
-        // FIXME: Expand the list of top level domains.
-        if (!(text.ends_with(".com") || text.ends_with(".net") || text.ends_with(".org"))) {
+        if (!(text.contains(".com") || text.contains(".net") || text.contains(".org"))) {
```
[the remark this answers is not in the document]
Comment (ardi0x01) [reply Userland/Applications/Browser/Tab.cpp:413]: yep, i'm just tested this, i will enhance and fix this, so do i need close this PR or continuing with new incoming commits?

[the remark this answers is not in the document]
Comment (EWouters) [reply Userland/Applications/Browser/Tab.cpp:413]: No need to close this, but you can mark it as 'Draft' while you're working on this.

File: Userland/Applications/Browser/Tab.cpp [hunk 2]
```diff
@@ -408,12 +408,17 @@ Optional<URL> Tab::url_from_location_bar(MayAppendTLD may_append_tld)
 
     StringBuilder builder;
     builder.append(text);
+
     if (may_append_tld == MayAppendTLD::Yes) {
-        // FIXME: Expand the list of top level domains.
```
Comment (stelar7) [inline Userland/Applications/Browser/Tab.cpp:412]: Maybe we can fetch this list from [IANA](https://data.iana.org/TLD/tlds-alpha-by-domain.txt)?

Comment (ardi0x01) [reply Userland/Applications/Browser/Tab.cpp:412]: Which list?

**Comment by ardi0x01** [comment]
about this enhancement, this is would be like if u pressing "test" in URL input then will be "https://www.google.com/search?q=test" depending on the search engine choose

**Comment by kleinesfilmroellchen** [comment]
Please squash the commits and address the reviews.

File: Userland/DevTools/HackStudio/LanguageServers/Cpp/Tests/complete_includes.cpp [hunk 3]
```diff
@@ -2,4 +2,3 @@
 #include <sys/cdef
 
 void foo() {}
-
```
Comment (MacDue) [inline Userland/DevTools/HackStudio/LanguageServers/Cpp/Tests/complete_includes.cpp:5]: This PR still includes a bunch of unrelated accidental changes, that should not be part of this PR. 
Including here...

File: Userland/Libraries/LibCpp/Tests/parser/inheritance.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
 class A : public SomeNamespace::B, private C
 {
     int x;
-};
+};
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/parser/inheritance.cpp:4]: and here

File: Userland/Libraries/LibCpp/Tests/parser/out-of-line.cpp [hunk 5]
```diff
@@ -7,4 +7,4 @@ class A
 
 bool A::foo() {
     return true;
-}
+}
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/parser/out-of-line.cpp:10]: and here

File: Userland/Libraries/LibCpp/Tests/parser/strace.cpp [hunk 6]
```diff
@@ -142,4 +142,4 @@ int main(int argc, char** argv)
     }
 
     return 0;
-}
+}
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/parser/strace.cpp:145]: and here

File: Userland/Libraries/LibCpp/Tests/preprocessor/macro1.cpp [hunk 7]
```diff
@@ -1,2 +1,2 @@
 #define ADD(x,y) x+y
-ADD(2,5);
+ADD(2,5);
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/preprocessor/macro1.cpp:2]: and here

File: Userland/Libraries/LibCpp/Tests/preprocessor/macro2.cpp [hunk 8]
```diff
@@ -1,3 +1,3 @@
 #define M(x) String {x + "lo"}
 
-M("he" + "l")
+M("he" + "l")
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/preprocessor/macro2.cpp:3]: and here

File: Userland/Libraries/LibCpp/Tests/preprocessor/macro3.cpp [hunk 9]
```diff
@@ -1,4 +1,3 @@
 #define M(x, y, z) x y = z;
 
 M(Vector, vec, ({1,2}))
-
```
Comment (MacDue) [inline Userland/Libraries/LibCpp/Tests/preprocessor/macro3.cpp:4]: and here

**Review by MacDue — CHANGES_REQUESTED** [review]

File: Userland/DevTools/HackStudio/LanguageServers/Cpp/Tests/complete_includes.cpp (hunk 3 above)
Comment (ardi0x01) [reply Userland/DevTools/HackStudio/LanguageServers/Cpp/Tests/complete_includes.cpp:5]: that and other files from this default from the branch, I undo these files change from prev commit, because prev commit they all not get the right format cause clang automation formatter in my local device

File: Userland/Applications/Browser/Tab.cpp [hunk 10]
```diff
@@ -407,10 +407,12 @@ Optional<URL> Tab::url_from_location_bar(MayAppendTLD may_append_tld)
     String text = m_location_box->text();
 
     StringBuilder builder;
-    builder.append(text);
+
+    String textUrl = g_search_engine.replace("{}", text);
```
Comment (creator1creeper1) [inline Userland/Applications/Browser/Tab.cpp:411]: "Use snake_case (all lowercase, with underscores separating words) for variable and function names."
See CodingStyle.md

File: Userland/Applications/Browser/Tab.cpp [hunk 11]
```diff
@@ -407,10 +407,12 @@ Optional<URL> Tab::url_from_location_bar(MayAppendTLD may_append_tld)
...
+    builder.append(textUrl);
+
     if (may_append_tld == MayAppendTLD::Yes) {
-        // FIXME: Expand the list of top level domains.
```
Comment (creator1creeper1) [inline Userland/Applications/Browser/Tab.cpp:412]: Has this actually been fixed?

Comment (ardi0x01) [reply Userland/Applications/Browser/Tab.cpp:412]: just enhance some condition, but I'm not fixed that above, I only enhance for if someone types a keyword on the URL tab, then the URL tab also automatically redirects into the search engine, depending on what search engine used

Comment (AtkinsSJ) [reply Userland/Applications/Browser/Tab.cpp:412]: Please don't remove a FIXME if you didn't fix the issue.

Comment (ardi0x01) [reply Userland/Applications/Browser/Tab.cpp:412]: I already restored that fix me comment :^)

File: Userland/Applications/Browser/Tab.cpp [hunk 12]
```diff
@@ -407,7 +407,10 @@ Optional<URL> Tab::url_from_location_bar(MayAppendTLD may_append_tld)
     String text = m_location_box->text();
 
     StringBuilder builder;
-    builder.append(text);
+
+    String text_url = g_search_engine.replace("{}", text);
+    builder.append(text_url);
```
Comment (timschumi) [inline Userland/Applications/Browser/Tab.cpp:412]: This looks like it will just wrap everything into a search query, even if it's a proper URL.

**Review by timschumi — CHANGES_REQUESTED** [review]
